src/test/java/io/vavr/collection/CollectionsTest.java

```diff
@@ -112,6 +113,27 @@ public void shouldUseMergeFunctionToHandleKeysCollisions() {
         assertThat(m.get("b").get().size()).isEqualTo(1);
     }
 
+    @Test
+    public void shouldUseMergeFunctionToHandleKeysCollisionsSortedMap() {
+        final Seq<Tuple2<String, Integer>> input =
+                Vector.of(Tuple.of("a",21), Tuple.of("a",21), Tuple.of("a",21), Tuple.of("bb",2));
+
+        Map<String, Integer> m = input.toSortedMap(
+                Comparator.comparing(String::length).reversed(),
+                Tuple2::_1,
+                Tuple2::_2,
+                Integer::sum);
+
+        assertThat(m.toVector()).isEqualTo(Vector.of(Tuple.of("bb", 2), Tuple.of("a", 63)));
+
+        Map<String, Integer> m2 = input.toSortedMap(
+                Tuple2::_1,
+                Tuple2::_2,
+                Integer::sum);
+
+        assertThat(m2.toVector()).isEqualTo(Vector.of(Tuple.of("a", 63), Tuple.of("bb", 2)));
+    }
+
```

Comment:
Please split the unit test. We strive for one assertion per test method.
I know that we often have more than one assert but the _testee_  should be called only one time per test.

In our case, we could create 4 test methods, two tests for each of the two `toSortedMap` conversion methods:

* converting an empty collection
* converting a non-empty collection

(I'm just thinking loud here: It is an implementation detail that the once conversion method delegates to the other. We don't expose that knowledge as public API.)

Reply:
That makes total sense: one test per use case. I added all tests for the `isEmpty` case as well.

I couldn't find any usages of a "spy" helper to check that `keyMapper`/`valueMapper`/`comparator`/`mergeFunction` are never called which I think makes the test a little less useful, but I'm happy to update the code if there's something that I can use to this effect.